### ISSUE TITLE
Can't build the doc

### ISSUE THREAD

**Issue by mlabo7** [sphinx-doc/sphinx#11132]
### Describe the bug

log file:
/tmp/sphinx-err-u2etzzf4.log

### How to Reproduce

happens in pipeline of:
https://readthedocs.org/projects/celery/builds/19173763/

### Environment Information

```text
v6.1.3
```

**Comment by jdillard** [comment]
If you need support I recommend the [support channels that ReadTheDocs recommends](https://docs.readthedocs.io/en/stable/support.html#site-support) or asking on StackOverflow: https://stackoverflow.com/questions/tagged/python-sphinx

**Comment by tk0miya** [comment]
The error comes from the extlinks extension. Could you check your markup around the extlink role, please? A caption inside the extlink role must contain `%s` exactly once.
```
Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/celery/envs/8005/lib/python3.9/site-packages/sphinx/ext/extlinks.py", line 103, in role
    title = caption % part
TypeError: not all arguments converted during string formatting
```

**Comment by tk0miya** [comment]
I found the reason. The error is raised when `extlinks` setting in conf.py contains a caption does not contain `%s`. Could you check your setting, please?

```
extlinks = {'issue': ('https://github.com/sphinx-doc/sphinx/issues/%s', 'issue')}
```

**Comment by papoteur-mga** [comment]
In our distribution (Mageia), we encountered this bug for several  packages.
Even if the `conf.py` is to blame, sphinx shouldn't fails in such a way. Either the string passed should be honoured, as it was the case before, or an explicit issue should be raised.
Changelog doesn't disclose any recent changes about `extlinks`

**Comment by befeleme** [comment]
This is also an issue for ~20 Fedora RPM packages, which was rather unexpected. I second the ask above.

**Comment by jdillard** [comment]
It looks like an exception was added for catching the lack of `%s`, but the logic may not always follow that path? https://github.com/sphinx-doc/sphinx/commit/7682353574e7c8b54330f2ce8f273a16816a6d02

**Comment by AA-Turner** [comment]
Duplicate of https://github.com/sphinx-doc/sphinx/issues/11094

@jdillard, the change was included in the 4.0 changelog:

https://github.com/sphinx-doc/sphinx/blob/075986c508381800c1a94f540a17963ed91abdf6/CHANGES#L1168

Should it also be retrospectively included in the 6.0 changelog?

A

**Comment by jdillard** [comment]
> The change was included in the 4.0 changelog. Should it also be retrospectively included in the 6.0 changelog?

Ah, I don't think so. I'm not sure if people are jumping from 3.0 to 6.0, but if they are, they should at least be checking each major version in the upgrade path.

**Comment by papoteur-mga** [comment]
> Even if the `conf.py` is to blame, sphinx shouldn't fails in such a way. Either the string passed should be honoured, as it was the case before, or an explicit issue should be raised. 

I think this is still true
